action.yml

```diff
@@ -80,7 +79,29 @@ runs:
 
         # Write the extracted version to a GitHub output variable
         # See <https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs-for-composite-actions>
-        echo "cargo-version=$(echo $version)" >> $GITHUB_OUTPUT
+        echo "cargo-version=$(echo $version)" >> "${GITHUB_OUTPUT}"
+
+    # If the workflow file's contents change, the new version's jobs should be considered distinct.
+    - name: Get workflow path
+      shell: bash
+      run: |
+        workflow_ref='${{ github.workflow_ref }}'
+        # Strip git ref suffix
+        workflow_abs_path="${workflow_ref%%@*}"
+        # Strip repository prefix
+        echo "workflow_path=${workflow_abs_path#'${{ github.repository }}/'}" >> "${GITHUB_ENV}"
+
+    - name: Determine cache group
+      id: cache-group
+      shell: bash
+      run: |
+        if [ -n '${{ inputs.cache-group }}' ]; then
+          # Use the `cache-group` input if it was provided.
+          echo 'cache-group=${{ inputs.cache-group }}' >> "${GITHUB_OUTPUT}"
+        else
+          # If not, fall back to a unique identifier for the workflow job.
+          echo 'cache-group=${{ hashFiles(env.WORKFLOW_PATH) }}-${{ github.job }}-${{ strategy.job-index }}' >> "${GITHUB_OUTPUT}"
```

Comment:
Not sure yet if we should rather put the workflow file hash at the end with fallbacks to be able to reuse the cache more often...
Maybe this should also not be overridden by the user provided group, I will take a look tomorrow